### PR TITLE
Add JSON output for git branch. Only when listing branches.

### PR DESCRIPTION
**Pull request by olsen232**
Something like:
```
{
    "sno.branch/v1": {
        "current": "newbie",
        "branches": {
            "master": {
                "commit": "abcdef1234567890",
                "abbrevCommit": "abcdef",
                "branch": "master",
                "upstream": {
                    "branch": "myremote/master",
                    "ahead": 0,
                    "behind": 0,
                },
            },
            "newbie": {
                "commit": "1234567890abcdef",
                "abbrevCommit": "123456",
                "branch": "newbie",
                "upstream": None,
            },
        },
    }
}
```

### CODE REVIEW

File: sno/branch.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import click
+import pygit2
+
+from .cli_util import MutexOption
+from .exec import execvp
+
+
+@click.command(context_settings=dict(ignore_unknown_options=True))
+@click.pass_context
+@click.option(
```
Comment (rcoup) [inline sno/branch.py:13]: Could we make this a single decorator we re-use?

eg:
```python
def sno_output_format_options(func):
  """ Apply Text & JSON output format options to a Click command """
  func = click.option("--text", ...)(func)
  func = click.option("--json", ...)(func)
  return func

@click.command(context_settings=dict(ignore_unknown_options=True))
@click.pass_context
@sno_output_format_options
@click.argument("args", nargs=-1, type=click.UNPROCESSED)
def branch(ctx, is_output_json, args):
  ...
```

File: sno/branch.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+)
+@click.argument("args", nargs=-1, type=click.UNPROCESSED)
+def branch(ctx, is_output_json, args):
+    """ List, create, or delete branches """
+    repo_path = ctx.obj.repo_path
+    repo = ctx.obj.repo
+
+    sargs = set(args)
+    if is_output_json:
+        if sargs - {"--list"}:  # sno branch --list --json is allowed.
```
Comment (rcoup) [inline sno/branch.py:38]:
```suggestion
        if "--list" not in sargs:  # sno branch --list --json is allowed.
```

File: sno/branch.py (hunk 1 above)
Comment (olsen232) [reply sno/branch.py:13]: Will do this in a separate PR

File: sno/branch.py (hunk 2 above)
Comment (olsen232) [reply sno/branch.py:38]: This has a different behavior. Here is what is allowed:
git branch --json
git branch --list --json 

Here is what is not allowed:
git branch -D delete_me --json

So I'm testing if sargs is empty, or just contains list. I'll rearrange to make it a bit clearer.

**Pull request merged**